tdp/cli/commands/plan/test_dag.py

```diff
@@ -5,20 +5,34 @@
 from pathlib import Path
 
 from click.testing import CliRunner
+from sqlalchemy import create_engine
 
-from tdp.cli.commands.init import init
 from tdp.cli.commands.plan.dag import dag
+from tdp.core.models import BaseModel, init_database
+from tdp.core.variables.cluster_variables import ClusterVariables
 
 
-def test_tdp_plan_dag(collection_path: Path, db_dsn: str, vars: Path):
+def test_tdp_plan_dag(
+    collection_path: Path,
+    db_dsn: str,
+    vars: Path,
+    create_cluster_variables: ClusterVariables,
+):
     base_args = [
         "--collection-path",
         collection_path,
         "--database-dsn",
         db_dsn,
     ]
+    # Create engine and initialize the database
+    engine = create_engine(db_dsn)
+    init_database(engine)
```

Comment:
You must not init the database like this. If the `assert` below or any lines of code raise an exception, the `drop_all` code will not run.

Reply:
It has been corrected with the second commit, However, now the engine is created twice, once through the fixture and once through the decorator `@database_dsn_option` which accompanies each cli function.